STX.Serialization.Providers.SystemTextJson/Services/Foundations/Serializations/SerializationService.cs

```diff
@@ -0,0 +1,62 @@
+﻿// ----------------------------------------------------------------------------------
+// Copyright (c) The Standard Organization: A coalition of the Good-Hearted Engineers
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using STX.Serialization.Providers.SystemTextJson.Brokers;
+using STX.Serialization.Providers.SystemTextJson.Models.Foundations.Serializations;
+
+namespace STX.Serialization.Providers.SystemTextJson.Services.Foundations.Serializations
+{
+    internal partial class SerializationService : ISerializationService
+    {
+        private readonly ISystemTextSerializationBroker systemTextSerializationBroker;
+
+        public SerializationService(ISystemTextSerializationBroker systemTextSerializationBroker)
+        {
+            this.systemTextSerializationBroker = systemTextSerializationBroker;
+        }
+
+        public ValueTask<TOutput> SerializeAsync<TInput, TOutput>(
+            TInput @object,
+            CancellationToken cancellationToken = default) =>
+            TryCatch(async () =>
+            {
+                ValidateInputIsNotNull(@object);
+                MemoryStream outputStream = new MemoryStream();
+
+                switch (typeof(TOutput))
+                {
+                    case Type _ when typeof(TOutput) == typeof(string):
+                        await Serialize(@object, outputStream, cancellationToken);
```

Comment:
could be after creating the MemoryStream.
It´s the same for all cases except the default case.
I understand you did it that way because you don´t waste resources/time when an error possibility happens. right?